### PR TITLE
Implement Text to Speech in the Story module

### PR DESCRIPTION
**Pull request by jeevithiesh**

### CODE REVIEW

**Comment by jeevithiesh** [comment]
PR checklist | Status
-- | --
Is code analysis task executed and issues resolved in the Java/Kotlin code? | Yes
Is code analysis task executed and issues resolved in the CPP/native code? | Yes
Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published? | Yes
Is code commenting done properly for the changes made? | Yes
Is correct variable naming convention followed? | Yes
Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files? | Yes
Are the message strings used in the application taken from strings.xml file? | NA
Are the permissions which are not required removed from AndroidManifest.java file? | NA
Is the uses-feature tag used for only must have feature in AndroidManifest.java file? | NA
Are the generated files correctly configured in .gitignore? | NA
Are the release signing configurations removed from the gradle files? | NA
Is commented code removed from all files? | Yes
Are updated images compressed (< 100 KB)? | NA
Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments? | Yes
Is size of the OBB correct? | Yes
Is OBB file version correctly set in DownloadExpansionFile.java file? | Yes
Is patch OBB file to be deployed? If yes, then are the settings correct in DownloadExpansionFile.java file? | NA

File: goa/Classes/story/StoryCoverPage.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "StoryCoverPage.hpp"
 #include "StoryPlaying.hpp"
 #include "QuestionHandler.h"
+#include <Managers/VoiceMoldManager.h>
```
Comment (ankitk20) [inline goa/Classes/story/StoryCoverPage.cpp]: Can you use " for include libraries?

File: goa/Classes/story/StoryCoverPage.cpp [hunk 2]
```diff
@@ -444,8 +446,23 @@ void StoryCoverPage::highlightWord(float time, cocos2d::ui::Text* text) {
 
 void StoryCoverPage::onEnterTransitionDidFinish() {
     Node::onEnterTransitionDidFinish();
-    this->scheduleOnce(schedule_selector(StoryCoverPage::narrateDialog), 1.0f);
-    this->scheduleOnce(schedule_selector(StoryCoverPage::highlightedNarrateWord), 1.0f);
+    // this->scheduleOnce(schedule_selector(StoryCoverPage::narrateDialog), 1.0f);
+    // this->scheduleOnce(schedule_selector(StoryCoverPage::highlightedNarrateWord), 1.0f);
```
Comment (ankitk20) [inline goa/Classes/story/StoryCoverPage.cpp]: Remove this if not required

File: goa/Classes/story/StoryPlaying.cpp [hunk 3]
```diff
@@ -728,27 +721,38 @@ void StoryPlaying::loadContentPageText() {
...
-    bool isSplitWordsEffectedLoaded = _loadedSplitWordsEffects.size() > 0;
-    
-    if(_soundEnabled.compare("true") == 0 && (isSplitWordsEffectedLoaded || (!_soundFile.empty() && FileUtils::getInstance()->isFileExist(_soundFile)))) {
-        
-        this->scheduleOnce(schedule_selector(StoryPlaying::narrateDialog), 1.0f);
-        this->scheduleOnce(schedule_selector(StoryPlaying::highlightedNarrateWord), 1.0f);
+    _jsonFile = "story/" + LangUtil::getInstance()->getLang() + "/" + _baseDir + ".json";
+    std::string jsonData = FileUtils::getInstance()->getStringFromFile(_jsonFile);
+    rapidjson::Document coverPageTextDocument;
+    if (false == coverPageTextDocument.Parse<0>(jsonData.c_str()).HasParseError()){
```
Comment (ankitk20) [inline goa/Classes/story/StoryPlaying.cpp]: Use ! instead of == false

File: goa/Classes/story/StoryPlaying.cpp [hunk 4]
```diff
@@ -728,27 +721,38 @@ void StoryPlaying::loadContentPageText() {
...
+        // document is ok
+        rapidjson::Value::MemberIterator M;
+        const char *key, *value;
+        int i = 0;
+        for (M = coverPageTextDocument.MemberBegin(); M != coverPageTextDocument.MemberEnd(); M++)
```
Comment (ankitk20) [inline goa/Classes/story/StoryPlaying.cpp]: Give a better name

File: goa/Classes/story/StoryPlaying.cpp [hunk 5]
```diff
@@ -1061,37 +1065,45 @@ void StoryPlaying::playSound(Ref* pSender, cocos2d::ui::Widget::TouchEventType e
...
-//                        CocosDenshion::SimpleAudioEngine::getInstance()->setEffectsVolume(1.0f);
-//                        CocosDenshion::SimpleAudioEngine::getInstance()->playBackgroundMusic(_soundFile.c_str(), false);
-//                    }
+                _jsonFile = "story/" + LangUtil::getInstance()->getLang() + "/" + _baseDir + ".json";
+                std::string jsonData = FileUtils::getInstance()->getStringFromFile(_jsonFile);
+                rapidjson::Document coverPageTextDocument;
+                if (false == coverPageTextDocument.Parse<0>(jsonData.c_str()).HasParseError())
+                {
+                    // document is ok
+                    rapidjson::Value::MemberIterator M;
```
Comment (ankitk20) [inline goa/Classes/story/StoryPlaying.cpp]: Give a better name

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: goa/Classes/story/StoryCoverPage.cpp (hunk 1 above)
Comment (jeevithiesh) [reply goa/Classes/story/StoryCoverPage.cpp]: I have changed it

File: goa/Classes/story/StoryPlaying.cpp (hunk 3 above)
Comment (jeevithiesh) [reply goa/Classes/story/StoryPlaying.cpp]: I have made changes

File: goa/Classes/story/StoryPlaying.cpp (hunk 5 above)
Comment (jeevithiesh) [reply goa/Classes/story/StoryPlaying.cpp]: I have made changes

File: goa/Classes/story/StoryCoverPage.cpp (hunk 2 above)
Comment (jeevithiesh) [reply goa/Classes/story/StoryCoverPage.cpp]: I have removed the comments

File: goa/Classes/story/StoryPlaying.cpp (hunk 4 above)
Comment (jeevithiesh) [reply goa/Classes/story/StoryPlaying.cpp]: I have made changes

File: goa/proj.android-studio/goa/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.maq.xprize.chimple.hindi"
     android:installLocation="preferExternal"
-    android:versionCode="13"
+    android:versionCode="15"
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/AndroidManifest.xml]: Change it 14

File: goa/proj.android-studio/goa/src/main/java/chimple/DownloadExpansionFile.java [hunk 7]
```diff
@@ -58,13 +58,13 @@
     public static final XAPKFile[] xAPKs = {
             new XAPKFile(
                     true, // true signifies a main file
-                    12, // the version of the main obb that is uploaded
-                    1450833724L  // the length of the file in bytes
+                    15, // the version of the main obb that is uploaded
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/chimple/DownloadExpansionFile.java]: Make it 14

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/TextToSpeechWrapper.java [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Wrapper for {@link TextToSpeech} with some handy test functionality.
+ */
+public class TextToSpeechWrapper {
+    private static final String MOCK_TTS_ENGINE = "com.android.cts.stub";
+    //public static final String FREE_TO_USE_TTS_ENGINE = "com.svox.pico";
+    //public static final String FREE_TO_USE_TTS_ENGINE = "com.example.android.ttsengine";
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/TextToSpeechWrapper.java]: Remove this

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/VoiceMoldManager.java [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+package org.cocos2dx.cpp;
+
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.HashMap;
+
+
+public class VoiceMoldManager {
+    // ----
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/VoiceMoldManager.java]: Remove this

File: goa/proj.android/jni/Application.mk [hunk 10]
```diff
@@ -1,4 +1,5 @@
 APP_STL := gnustl_static
+APP_DEBUG :=1
```
Comment (ankitk20) [inline goa/proj.android/jni/Application.mk]: Space after =

File: goa/proj.android/jni/Application.mk [hunk 11]
```diff
@@ -1,4 +1,5 @@
...
 
 APP_CPPFLAGS := -frtti -DCC_ENABLE_CHIPMUNK_INTEGRATION=1 -std=c++11 -fsigned-char
```
Comment (ankitk20) [inline goa/proj.android/jni/Application.mk]: Remove ogg files that are not required

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: goa/proj.android-studio/goa/src/main/AndroidManifest.xml (hunk 6 above)
Comment (jeevithiesh) [reply goa/proj.android-studio/goa/src/main/AndroidManifest.xml]: I have made the changes

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/TextToSpeechWrapper.java (hunk 8 above)
Comment (jeevithiesh) [reply goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/TextToSpeechWrapper.java]: I have removed the comments

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/VoiceMoldManager.java (hunk 9 above)
Comment (jeevithiesh) [reply goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/VoiceMoldManager.java]: I have removed the comment

File: goa/proj.android/jni/Application.mk (hunk 10 above)
Comment (jeevithiesh) [reply goa/proj.android/jni/Application.mk]: I have added a space

File: goa/proj.android/jni/Application.mk (hunk 11 above)
Comment (jeevithiesh) [reply goa/proj.android/jni/Application.mk]: I have removed the audio files that are not required

**Review by ankitk20 — DISMISSED** [review]

File: goa/Classes/menu/MenuContext.cpp [hunk 12]
```diff
@@ -1184,7 +1164,7 @@ void MenuContext::launchGameFinally(std::string gameName) {
 			Director::getInstance()->replaceScene(BasicLetterCase::createScene());
 		}
 		else{
-            CCLOG("Failed starting scene: %s", gameName.c_str());
+
```
Comment (mathewnik90) [inline goa/Classes/menu/MenuContext.cpp]: Why do you need blank else?

**Review by mathewnik90 — CHANGES_REQUESTED** [review]

File: goa/Classes/menu/MenuContext.cpp (hunk 12 above)
Comment (jeevithiesh) [reply goa/Classes/menu/MenuContext.cpp]: I have removed the else

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**